conjureplugin/param.go

```diff
@@ -41,4 +41,8 @@ type ConjureProjectParam struct {
 	AcceptFuncs bool
 	// Publish specifies whether or not this Conjure project should be included in the "publish" operation.
 	Publish bool
+	// SkipDeleteGeneratedFiles skips cleanup of old generated files before regeneration.
+	// When false (default), deletes all Conjure-generated files in the output directory tree before regenerating.
+	// When true, preserves v1 behavior (no cleanup).
+	SkipDeleteGeneratedFiles bool
```

Comment:
Can we add an integration test that verifies the behavior of `SkipDeleteGeneratedFiles` being true vs. false?